reporting/src/main/java/org/wso2/testgrid/reporting/PerformanceResultProcessor.java

```diff
@@ -0,0 +1,240 @@
+        if (!copyOfDividers.isEmpty()) {
+            //if this is the last divider in the list it is a data section containing tables
+            boolean isDataSection = (copyOfDividers.size() == 1);
+            String divider = copyOfDividers.pop();
+            //filter distinct sections of data for each divider
+            int index = topics.indexOf(divider);
+            List<String> dividerSubSections = section.stream()
+                    .map(strings -> strings.get(index))
+                    .distinct()
+                    .collect(Collectors.toList());
+
+            if (isDataSection) {
+                DividerSection dividerSection = new DividerSection();
+                dividerSection.setData(divider);
+                //create a new data section
+                DataSection dataSection = new DataSection();
+                List<PerformanceTable> performanceTableList = new ArrayList<>();
+                //create performance data tables for the last divider section
+                dividerSubSections.forEach(s -> {
+                    List<List<String>> collect1 = section.stream()
+                            .filter(strings -> strings.get(index).equals(s))
+                            .collect(Collectors.toList());
+                    PerformanceTable table = createPerformanceTable(topics, collect1, resultFormatter);
+                    table.setDescription(s);
+                    performanceTableList.add(table);
+                });
+                dataSection.setPerformanceTableList(performanceTableList);
+                dataSection.setDescription(divider);
+                dividerSection.setChildSections(null);
+                dividerSection.setDataSection(dataSection);
+                return dividerSection;
+            } else {
+                //if there are more divider sections it means data needs to be further divided
+                List<DividerSection> dividerSections = new ArrayList<>();
+                dividerSubSections.forEach(s -> {
+                    List<List<String>> collect1 = section.stream()
+                            .filter(strings -> strings.get(index).equals(s))
+                            .collect(Collectors.toList());
+                    //recursively set divider sections
+                    DividerSection dividerSection = this.getSectionData(copyOfDividers, collect1, topics
+                            , resultFormatter);
+                    dividerSection.setData(s);
+                    dividerSections.add(dividerSection);
+                });
+                DividerSection dividerSection = new DividerSection();
+                dividerSection.setData(divider);
+                dividerSection.setChildSections(dividerSections);
+                dividerSection.setDataSection(null);
+                return dividerSection;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Creates the performance table given the data and formatter data.
```

Comment:
Explain the 'why' aspect here please. For ex. Why you create a performance table like this?